packages/grafana-data/src/events/types.ts

```diff
@@ -18,11 +18,21 @@ export abstract class BusEventBase implements BusEvent {
   readonly type: string;
   readonly payload?: any;
   readonly origin?: EventBus;
+  tags?: Set<string>;
 
   constructor() {
     //@ts-ignore
     this.type = this.__proto__.constructor.type;
   }
+
+  /**
+   * @internal
+   * Tag event for finer-grained filtering in subscribers
+   */
+  setTags(tags: string[]) {
+    this.tags = new Set(tags);
+    return this;
+  }
```

Comment:
I think might be cleaner to add this specifically to DataHoverEvent.

Something like this? 

```
export class DataHoverEvent extends BusEventWithPayload<DataHoverPayload> {
  static type = 'data-hover';

  constructor(payload: DataHoverPayload, public tags?: { uPlot: boolean }) {
    super(payload);
  }
}
```

Reply:
alternative is to add the tags/flags on the payload (we would need to add payload to DataHoverClearEvent)

Reply:
hmm, not sure. i kinda like having a generic fine grained filter-ability on all events? also having `uPlot` in a shared interface feels weird / too specific.

Reply:
I can't see a really strong argument for not having be just typed property on the payload

Reply:
No strong opinion, but I think I prefer the `setTags` interface rather than modifying the payload type(s). It feels like it separates nicely the event semantics from the event/classification. +1 to @leeoniya comment on leaking uPlot specifics to a general interaction event payload.

Reply:
but it feels like such a specific thing to these data events. 

what about 
```ts
const hoverEvent = new DataHoverEvent(payload, ["uplot"]);
```

the setTags that mutate an event feels so wrong, should be part of the constructor

Reply:
> const hoverEvent = new DataHoverEvent(payload, ["uplot"]);

yeah, this feels good I think

Reply:
sounds good, will update 👍

Reply:
actually, thinking more about this, are we sure we want this API?

if we move this to the constructor, we lose the ability to ever move it up to the more general `BusEventBase`, due to now-required and incompatible constructor signatures.

![image](https://github.com/grafana/grafana/assets/43234/b1d8a97b-62f8-42ea-bc72-318138ff6987)

additionally, what's the benefit of only allowing data hover/clear events to be tagged?

i see another event below these two that could possibly need the same treatment in the future (where uPlot can efficiently sync the selection of 2k scatter points) , and who knows what other event types can use this?

![image](https://github.com/grafana/grafana/assets/43234/c6eae72f-5bfa-4492-be80-0c253659e94b)

i'm still in the camp that the current arrangement has a lot of benefits and no real drawbacks.

Reply:
Good point about the constructor @leeoniya.  I think the `setTags` approach is fine and indeed more future-friendly.